ui/main/src/app/services/processes.service.ts

```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
         };
     }
 
+    public findServiceForProcess(processId : string) : string {
+        for (let group of this.processGroups) {
+            if (group.processes.find(process => process == processId))
+                return group.id;
+        }
+        return '';
+    }
+
+    public makeProcessesDropdownListPerServices(): Map<any, any> {
+        let processesDropdownListPerServices = new Map();
+
+        this.getAllProcesses().forEach(process => {
+
+            const service = this.findServiceForProcess(process.id);
+            if (service != '') {
+                let processes = (!! processesDropdownListPerServices.get(service) ? processesDropdownListPerServices.get(service) : []);
+                processes.push({id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}`});
+                processesDropdownListPerServices.set(service, processes);
+            }
+        });
+        return processesDropdownListPerServices;
+    }
+
+    public makeProcessesWithoutServiceDropdownList(): any[] {
+        let processesWithoutServiceDropdownList = [];
+
+        this.getAllProcesses().forEach(process => {
+            const service = this.findServiceForProcess(process.id);
+            if (service == '')
+                processesWithoutServiceDropdownList.push({ id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}` });
+        });
+        return processesWithoutServiceDropdownList;
+    }
+
+    public loadAllProcessesPerServices(object: {processesDropdownListPerServices: Map<any, any>;
+                                                processesWithoutServiceDropdownList: any[];
+                                                serviceDropdownList: any[];}): void {
+
+        object.processesDropdownListPerServices = this.makeProcessesDropdownListPerServices();
+        object.processesWithoutServiceDropdownList = this.makeProcessesWithoutServiceDropdownList();
+        if (object.processesWithoutServiceDropdownList.length > 0)
+            object.serviceDropdownList.push({ id: '--', itemName: "service.defaultLabel" });
+
+        const services = Array.from(object.processesDropdownListPerServices.keys());
+        services.forEach(service => object.serviceDropdownList.push({ id: service, itemName: service }));
+    }
+
+    public findServiceLabelForProcess(processId : string) : string {
+        const serviceId = this.findServiceForProcess(processId);
+        return (!! serviceId && serviceId != '') ? serviceId : "service.defaultLabel";
+    }
+
+    public makeMapOfProcessStates() : Map<string, TypeOfStateEnum> {
+        let mapOfProcessStates = new Map();
+
+        for (let process of this.processes) {
+            for (let state in process.states)
+                mapOfProcessStates.set(process.id + '.' + state, process.states[state].type);
+        }
+        return mapOfProcessStates;
+    }
```

Comment:
For all the file :  replace "DropdownList" with "List" as the method are more generic and could be used for other purposes than dropdown list

Reply:
Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01